src/common/ApprovalsChip.tsx

```diff
@@ -1,14 +1,31 @@
+import React, { useState, MouseEvent } from 'react'
+
 import Close from '@mui/icons-material/CloseTwoTone'
-import DoneAll from '@mui/icons-material/DoneAllTwoTone'
 import Done from '@mui/icons-material/DoneTwoTone'
+import DoneAll from '@mui/icons-material/DoneAllTwoTone'
 import Chip from '@mui/material/Chip'
+import Menu from '@mui/material/Menu'
+import List from '@mui/material/List'
+import ListItem from '@mui/material/ListItem'
+import ListItemIcon from '@mui/material/ListItemIcon'
+import ListItemText from '@mui/material/ListItemText'
+import AccessTime from '@mui/icons-material/AccessTime'
+import DownArrow from '@mui/icons-material/KeyboardArrowDown'
+import UpArrow from '@mui/icons-material/KeyboardArrowUp'
+import Stack from '@mui/material/Stack'
+
 import useTheme from '@mui/styles/useTheme'
-import React from 'react'
-import { Theme } from '../theme'
+import { Theme } from '../../src/theme'
 
 export default function ApprovalsChip({ approvals }: { approvals: any }) {
-  const numApprovals = approvals.filter((e: string) => e === 'Accepted').length
-  const totalApprovals = approvals.length
+  Object.keys(approvals).forEach((key) => (approvals[key] === undefined ? delete approvals[key] : {}))
+
+  const approvalOutcomes = [approvals?.managerResponse, approvals?.reviewerResponse].filter(Boolean)
+  const numApprovals = approvalOutcomes.filter((e: string) => e === 'Accepted').length
```

Comment:
All of these in-line calculations (`numApprovals`, `totalApprovals`, `open`) will be calculated on every single render. I tend to prefer to memoize them so that they're only recalculated when they need to be (when one of the values in the dependency array changes). E.g:
```
const numApprovals = useMemo(() => {
  approvalOutcomes.filter((e: string) => e === 'Accepted').length
}, [approvalOutcomes])
```

There is a slight overhead to using `useMemo`, but I find it's usually worth playing it safe and using it for anything mildly complex.